### PR TITLE
DOC: Add missing bracket

### PR DESCRIPTION
**Pull request by Qiyu8**
I noticed that the close parenthesis was missing in a lot of places, which generate:
- example cases that could not be run.
- misleading document.
so I write a script to auto-detect missing bracket:
```
import os

class Stack(object):
    def __init__(self):
        self.stack=[]
    def isEmpty(self):
        return self.stack==[]
    def push(self,item):
        self.stack.append(item)
    def pop(self):
        if self.isEmpty():
            raise IndexError
        return self.stack.pop()
    def peek(self):
        return self.stack[-1]
    def size(self):
        return len(self.stack)

def scanSingleFile(filePath):
    with open(filePath, 'r', encoding ='utf-8') as f:
        contents = f.readlines()
        count = 1
        stack=Stack()
        for line in contents:
            for ch in line:
                if ch == '(':
                    stack.push(count)
                elif ch == ')':
                    if not stack.isEmpty():
                        stack.pop()
            count+=1
        if not stack.isEmpty():
            formatPath = filePath.replace(r'\/'.replace(os.sep,''), os.sep)
        while not stack.isEmpty():
            print(formatPath+":"+str(stack.pop()))

def scanMissingBracket(scanpath):
    for root, dirs, files in os.walk(scanpath):
        for file in files:
            filePath = os.path.join(root,file)
            if os.path.splitext(filePath)[1] in ['.py','.c','.h','.rst','.txt']:
                scanSingleFile(filePath)

if __name__ == "__main__":
    scanMissingBracket("./numpy")
```
All of the results has been carefully filtered. but only `numpy` directory is scanned, should I scan the `doc `directory too?

### CODE REVIEW

**Comment by seberg** [comment]
Nice, looks good to me. The lapack-lite changes are probably unnecessary or not helpful, since that is not our code (it is just copied in).

File: numpy/linalg/lapack_lite/f2c_c_lapack.c [hunk 1]
```diff
@@ -12813,7 +12813,7 @@ static real c_b2435 = .5f;
 
     RWORK  (workspace) REAL array, dimension at least
            (9*N + 2*N*SMLSIZ + 8*N*NLVL + 3*SMLSIZ*NRHS +
-           MAX( (SMLSIZ+1)**2, N*(1+NRHS) + 2*NRHS ),
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/f2c_c_lapack.c:12816]: This is generated code, and should not be charged

File: numpy/linalg/lapack_lite/f2c_z_lapack.c [hunk 2]
```diff
@@ -12881,7 +12881,7 @@ static doublereal c_b2435 = .5;
 
     RWORK  (workspace) DOUBLE PRECISION array, dimension at least
            (9*N + 2*N*SMLSIZ + 8*N*NLVL + 3*SMLSIZ*NRHS +
-           MAX( (SMLSIZ+1)**2, N*(1+NRHS) + 2*NRHS ),
```
Comment (eric-wieser) [inline numpy/linalg/lapack_lite/f2c_z_lapack.c:12884]: Same here

**Comment by Qiyu8** [comment]
Thanks for the review, Both `numpy` and `doc `directory have been scanned.

**Comment by seberg** [comment]
Thanks, lets put this in.

**Pull request merged**